solr/core/src/java/org/apache/solr/util/ConcurrentLRUCache.java

```diff
@@ -552,15 +553,14 @@ private void markAndSweepByCacheSize() {
           // this loop so far.
           queue.myMaxSize = sz - lowerWaterMark - numRemoved;
           while (queue.size() > queue.myMaxSize && queue.size() > 0) {
-            @SuppressWarnings({"rawtypes"})
-            CacheEntry otherEntry = queue.pop();
+            CacheEntry<K,V> otherEntry = queue.pop();
             newOldestEntry = Math.min(otherEntry.lastAccessedCopy, newOldestEntry);
           }
           if (queue.myMaxSize <= 0) break;
 
           Object o = queue.myInsertWithOverflow(ce);
           if (o != null) {
-            newOldestEntry = Math.min(((CacheEntry)o).lastAccessedCopy, newOldestEntry);
+            newOldestEntry = Math.min(((CacheEntry<?,?>)o).lastAccessedCopy, newOldestEntry);
```

Comment:
... would allow
```suggestion
            newOldestEntry = Math.min(o.lastAccessedCopy, newOldestEntry);
```

cast removal here I think (but the UI won't allow that to be suggested in one comment)